File: src/task/src/control_publisher.cpp
#include "ros/ros.h"
#include "std_msgs/String.h"
#include <iostream>

using namespace std;

int main(int argc, char **argv){
    cout << "control_publisher.cpp begin" << endl;
    ros::init(argc, argv, "control");
    ros::NodeHandle nh;
    
    ros::Publisher pub = nh.advertise<std_msgs::String>("control_topic", 100);
    ros::Rate rate(0.5);

    std_msgs::String msg;
    msg.data = "start";

    int count = 0;
    while(ros::ok()){
        ROS_INFO("%s", msg.data.c_str());
        pub.publish(msg);

        if(count>5){
            msg.data = "shutdown";
        }
        count += 1;

        ros::spinOnce();
        rate.sleep();
    }
    return 0;
}

File: src/task/src/control_publisher.py
#!/usr/bin/env python2
#coding:utf-8

import rospy
from std_msgs.msg import String

if __name__ == '__main__':
    rospy.init_node('control')
    pub = rospy.Publisher('control_topic', String, queue_size=1)

    rate = rospy.Rate(10)
    count = 0
    while not rospy.is_shutdown():
        if count>5:
            pub.publish("shutdown")
        else:
            pub.publish("start")
        count += 1
        rate.sleep()
